projects/packages/videopress/src/class-access-control.php

```diff
@@ -0,0 +1,355 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+/**
+ * VideoPress Access Control.
+ *
+ * @package automattic/jetpack-videopress
+ */
+
+namespace Automattic\Jetpack\VideoPress;
+
+use Automattic\Jetpack\Connection\Client;
+use VIDEOPRESS_PRIVACY;
+
+/**
+ * VideoPress video access control utilities.
+ *
+ * Note: this is also being used on WordPress.com.
+ * Use IS_WPCOM checks for functionality that is specific to WPCOM/Jetpack.
+ */
+class Access_Control {
+
+	/**
+	 * Singleton Access_Control instance.
+	 *
+	 * @var Access_Control
+	 **/
+	private static $instance = null;
+
+	/**
+	 * Guid to subscription plan, store, for when used inline on a page.
+	 *
+	 * @var array
+	 */
+	private $guids_to_subscriptions = array();
+
+	/**
+	 * Set that this guid is controlled by a subscription.
+	 *
+	 * @param string     $guid            The guid to set.
+	 * @param string|int $subscription_id The subscription to set.
+	 *
+	 * @return Access_Control
+	 */
+	public function set_guid_subscription( $guid, $subscription_id ) {
+		$this->guids_to_subscriptions[ $guid ] = $subscription_id;
+		return $this;
+	}
+
+	/**
+	 * Get the subscription for a guid.
+	 *
+	 * @param string $guid The guid to get.
+	 *
+	 * @return string|int|false
+	 */
+	public function get_subscription_plan_id( $guid ) {
+		return isset( $this->guids_to_subscriptions[ $guid ] ) ? $this->guids_to_subscriptions[ $guid ] : false;
+	}
+
+	/**
+	 * Get the singleton instance.
+	 *
+	 * @return AJAX
+	 */
+	public static function instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Determines if Jetpack Memberships are available.
+	 *
+	 * @return bool
+	 */
+	private function jetpack_memberships_available() {
+		return class_exists( '\Jetpack_Memberships' );
+	}
+
+	/**
+	 * Determines if Jetpack Subscriptions are available.
+	 *
+	 * @return bool
+	 */
+	private function jetpack_subscriptions_available() {
+		if ( function_exists( '\Automattic\Jetpack\Extensions\Premium_Content\subscription_service' ) ) {
+			return true;
+		}
+
+		if ( ! defined( 'JETPACK__PLUGIN_DIR' ) ) {
+			return false;
+		}
+
+		$subscription_service_file_path = JETPACK__PLUGIN_DIR . 'extensions/blocks/premium-content/_inc/subscription-service/include.php';
+		if ( ! file_exists( $subscription_service_file_path ) ) {
```

Comment:
It looks like they are checking if premium content block is loaded?

Reply:
@jeherve @millerf yes, to my understanding there are situations it would not exist. Since all gating/subscription classes are part of only the Jetpack plugin, any other flavor (specifically Jetpack VideoPress plugin) would not contain those.